MediaGalleryMetadata/Model/AddXmpMetadata.php

```diff
@@ -36,9 +36,21 @@ public function execute(string $data, MetadataInterface $metadata): string
             $xml->registerXPathNamespace($prefix, $url);
         }
 
-        $this->setValueByXpath($xml, self::XMP_XPATH_SELECTOR_TITLE, $metadata->getTitle());
-        $this->setValueByXpath($xml, self::XMP_XPATH_SELECTOR_DESCRIPTION, $metadata->getDescription());
-        $this->updateKeywords($xml, $metadata->getKeywords());
+        if (empty($metadata->getTitle())) {
+            $this->deleteValueByXpath($xml, self::XMP_XPATH_SELECTOR_TITLE);
+        } elseif ($metadata->getTitle() !== null) {
+            $this->setValueByXpath($xml, self::XMP_XPATH_SELECTOR_TITLE, $metadata->getTitle());
+        }
```

Comment:
The sequence of conditions needs to be reversed (for all fields)
```suggestion
        if ($metadata->getTitle() === null) {
            $this->deleteValueByXpath($xml, self::XMP_XPATH_SELECTOR_TITLE);
        } else {
            $this->setValueByXpath($xml, self::XMP_XPATH_SELECTOR_TITLE, $metadata->getTitle());
        }
```